### ISSUE TITLE
Docker integration?

### ISSUE THREAD

**Issue by andreis** [codegangsta/gin#35]
It's pretty straightforward to ship a Go app using e.g. [google/golang](https://registry.hub.docker.com/u/google/golang/), but I'm not sure how to setup an environment that works with gin. Any ideas?

**Comment by caiges** [comment]
While using the `golang:1.4-wheezy` and a simple martini example within a docker container, I get `http: proxy error: dial tcp 127.0.0.1:3001: connection refused`.

**Comment by caiges** [comment]
Here's the dockerfile and example code I'm using:

https://gist.github.com/caiges/3527b9213ce841f511fe

and the command I'm using:

gin --path /opt/helloworld /opt/helloworld/main.go

**Comment by Raul-diffindo** [comment]
When I integrate Gin within Docker I receive this error message:

"/go/bin/gin: 1: /go/bin/gin: Syntax error: "(" unexpected"

Any ideas?

**Comment by caiges** [comment]
Hmm, haven't observed that behavior. Can you link your code?

**Comment by Raul-diffindo** [comment]
It´s a concept test only:

[server.go]

package main

import "github.com/go-martini/martini"

func main() {
m := martini.Classic()
m.Get("/", func() string {
return "Hello world!"
})
m.Run()
}

[Dockerfile]

FROM ubuntu:13.10

RUN apt-get update
RUN apt-get install -y build-essential mercurial git subversion wget curl
bzr

RUN DEBIAN_FRONTEND=noninteractive apt-get -y install golang

RUN mkdir /go
RUN export GOPATH=/go
ENV GOPATH /go
RUN export PATH=$PATH:$GOPATH/bin
ENV PATH $PATH:$GOPATH/bin

WORKDIR /go

RUN go get github.com/tools/godep
RUN go get github.com/go-martini/martini
RUN go get github.com/codegangsta/gin

CMD go install lifes.com && gin --path src/lifes.com/ src/
lifes.com/server.go



2015-01-27 18:18 GMT+01:00 Caige Nichols <notifications@github.com>:

**Comment by caiges** [comment]
Hmm, might be something in the lifes.com repo you're using. I slightly modified your [Dockerfile](https://gist.github.com/caiges/28701c5633294548fed0), built with `docker built -t testapp` and was able to run `docker run testapp gin --path /usr/src/app /usr/src/app/main.go`

**Comment by Raul-diffindo** [comment]
Yeah! Thanks caiges! now this part is working but now I received the following error message:

$ docker run -p 3000:3000 lifes-server gin --path src/lifes.com/ src/lifes.com/server.go 
[gin] listening on port 3000
2015/01/27 18:33:52 http: proxy error: dial tcp 127.0.0.1:3001: connection refused

Any ideas?

Thank you so much!

**Comment by ntboes** [comment]
I just released my Docker Container which runs gin. https://github.com/ntboes/docker-golang-gin https://registry.hub.docker.com/u/ntboes/golang-gin/

Usage:
```
docker run --name some-instance-name -v $(pwd):/go/src/app ntboes/golang-gin -p 4000
```

**Comment by HermannBjorgvin** [comment]
@Raul-diffindo I had the same error when I was running without the CGO_ENABLED=0 flag
